Cython/Utility/AsyncGen.c

```diff
@@ -216,15 +218,22 @@ __Pyx_async_gen_init_hooks(__pyx_PyAsyncGenObject *o)
 
     o->ag_hooks_inited = 1;
 
+#if CYTHON_COMPILING_IN_CPYTHON
```

Comment:
I'm not completely sure, but I wonder if `CYTHON_USE_TYPE_SLOTS` would be the more appropriate check here (and later)

Reply:
I am not sure, I thought `CYTHON_USE_TYPE_SLOTS` is for the documented dunder slots like `tp_str` -> `__str__` as generated by the [`typeslots.py`](https://github.com/python/cpython/blob/0449ffe3a4ddf03367a5ee3d943c89f442b7b407/Objects/typeslots.py#L2) code generator.

Reply:
... which generates [these slot functions](https://github.com/python/cpython/blob/0449ffe3a4ddf03367a5ee3d943c89f442b7b407/Objects/typeslots.inc#L1)

Reply:
`CYTHON_USE_TYPE_SLOTS` is for code that directly uses the `tp_*` slot functions of types, not for attributes on the thread state. There's `CYTHON_FAST_THREAD_STATE`, but I don't think that applies here, either. That guard controls the way we get at the thread state, more that what we do with it (although that might still depend on how fast we can get our hands on it when we need it).

If this it PyPy specific code, why not use `CYTHON_COMPILING_IN_PYPY`?